library/core/functions.validation.php

```diff
@@ -528,6 +528,27 @@ function validateLength($value, $field) {
     }
 }
 
+if (!function_exists('validateRawLength')) {
+    /**
+     * Validate that a string is not too long when most of the formatting and meta data is removed.
+     *
+     * @param mixed $value The value to validate.
+     * @param object $field The field information that contains the maximum length for the {@link $value}.
+     * @param array $post The Post array from the form.
+     * @return bool|string
+     */
+    function validateRawLength($value, $field, $post) {
+        $format = $post['Format'] ?? '';
+        $stringLength = Gdn::FormatService()->getVisibleTextLength($value, $format);
+        $diff = $stringLength - $field->Length;
+        if ($diff <= 0) {
+            return true;
+        } else {
+            return sprintf(t('ValidateLength'), t($field->Name), $diff);
+        }
+    }
+}
+
```

Comment:
Are we sure we want to introduce new global function?

Reply:
I'm not sure, but I think it was originally done this way because it's the way 98% of our validation functions are implemented. However, the other 2% demonstrate we do not need a global function. As long as the "rule" argument passed to `addRule` is a callable, it'll be invoked using the value. We can see an example over in `VanillaSettingsController`.

https://github.com/vanilla/vanilla/blob/e6df634ff288e1d9e5a82092a0e2230a835ab1bb/applications/vanilla/controllers/class.vanillasettingscontroller.php#L162